demos/exit/README.md

```diff
@@ -0,0 +1,38 @@
+# Exit
+
+In this demo we are going to show that threads can exit normally like C functions. So we can use functions like `thread_join` to let a thread to wait for some threads to exit. 
+
```

Comment:
I think you should note here the difference between exiting the kernel and a thread exiting. (or finishing...the terminology overlap gets a bit weird)

Something like:
"This is in contrast to exiting using `exit()` which exits the whole system."

Also I think `thread_start` should be one of the introduced functions. Noting that:
* threads start here before they run their worker function
* by doing so we catch it's return and set its state to finished and go back to the kernel

You don't need to mention it in the walkthrough but I think it's good to include a bit of the detail of how we handle threads returning.

Reply:
> Also I think thread_start should be one of the introduced functions.

Thanks for pointing out this. I didn't notice that before, and I think it is really helpful to understand the change of the state in this demo.

Reply:
> Also I think thread_start should be one of the introduced functions. 

Thanks for pointing out this. I didn't even notice that before...